aggregator/rpc_server.go

```diff
@@ -40,25 +41,50 @@ func (agg *Aggregator) startServer() error {
 	return nil
 }
 
+func (agg *Aggregator) verifyOperatorIdRegistered(operatorId eigentypes.OperatorId, messageDigest [32]byte, signature bls.Signature) (bool, error) {
+	operatorPubkeys, ok := agg.operatorRegistrationsService.GetOperatorPubkeysById(context.Background(), operatorId)
+	if !ok {
+		return false, OperatorNotPartOfTaskQuorum400
+	}
+
+	ok, err := signature.Verify(operatorPubkeys.G2Pubkey, messageDigest)
+	if err != nil {
+		return false, UnknownErrorWhileVerifyingSignature400
+	}
+
+	return ok, nil
+}
+
 // rpc endpoint which is called by operator
 // reply doesn't need to be checked. If there are no errors, the task response is accepted
 // rpc framework forces a reply type to exist, so we put bool as a placeholder
 func (agg *Aggregator) ProcessSignedCheckpointTaskResponse(signedCheckpointTaskResponse *messages.SignedCheckpointTaskResponse, reply *bool) error {
 	agg.logger.Info("Received signed task response", "response", fmt.Sprintf("%#v", signedCheckpointTaskResponse))
-
-	taskIndex := signedCheckpointTaskResponse.TaskResponse.ReferenceTaskIndex
 	taskResponseDigest, err := signedCheckpointTaskResponse.TaskResponse.Digest()
 	if err != nil {
 		agg.logger.Error("Failed to get task response digest", "err", err)
 		return TaskResponseDigestNotFoundError500
 	}
 
+	operatorId := signedCheckpointTaskResponse.OperatorId
+	signature := signedCheckpointTaskResponse.BlsSignature
+	ok, err := agg.verifyOperatorIdRegistered(operatorId, taskResponseDigest, signature)
+	if err != nil {
+		agg.logger.Error("Failed to verify message", "err", err)
+		return err
```

Comment:
IMO we should either add a new error or just return it as SignatureVerificationFailed anyway. Wdyt? Applies to all similar cases

Reply:
Agree mostly for consistency

Reply:
I now moved this check to `Aggregator` directly, and the error is mapped inside `RpcServer`